skyllh/analyses/i3/publicdata_ps/utils.py

```diff
@@ -253,3 +310,29 @@ def create_energy_cut_spline(ds, exp_data, spl_smooth):
         sindec_centers, min_log_e, k=2, s=spl_smooth)
 
     return spline
+
+
+def tdm_field_func_psi(
+        tdm,
+        shg_mgr,
+        pmm):
+    """TDM data field function to calculate the opening angle between the
+    source positions and the event's reconstructed position.
+    """
+    (src_idxs, evt_idxs) = tdm.src_evt_idxs
+
+    ra = np.take(tdm.get_data('ra'), evt_idxs)
+    dec = np.take(tdm.get_data('dec'), evt_idxs)
+
+    src_array = tdm.get_data('src_array')
+    src_ra = np.take(src_array['ra'], src_idxs)
+    src_dec = np.take(src_array['dec'], src_idxs)
+
+    psi = angular_separation(
+        ra1=ra,
+        dec1=dec,
+        ra2=src_ra,
+        dec2=src_dec,
+        psi_floor=10**-5.95442953)
```

Comment:
The given `psi_floor` value here is only needed for the correct KDE PDFs evaluation, due to the way how photosplines were built. I think that for analyses which are using Gaussian spatial term one could set this to 0 by default. One way to make it more generic would be to provide `get_tdm_field_func_psi` function `with psi_floor=0` kwarg, which would set the `psi_floor` and return `tdm_field_func_psi`.

Reply:
In SkyLLH as it is right now, the `psi_func` to compute the angular distance between the events and the source is in the analysis scripts, e.g. `time_integrated_ps.py` and `time_dependent_ps.py`.

In that function, no psi_floor is being applied: https://github.com/icecube/skyllh/blob/009c6a7cbd0417d13c9cbcf17bbca4e0fa3feb9f/skyllh/analyses/i3/publicdata_ps/time_integrated_ps.py#L90-L117